build.bash

```diff
@@ -9,9 +9,9 @@
 # Outputs compiled program in current directory.
 # Default file name is 'ecaddy'.
 
-set -e
+set -euo pipefail
 
-: ${output_filename:="$1"}
+: ${output_filename:="${1:-}"}
 : ${output_filename:="ecaddy"}
```

Comment:
A substitution expression inside the other is frowned upon because it becomes hard to read.